### PR TITLE
CORE-5152 Create Gateway Processor

### PR DESCRIPTION
**Pull request by williamvigorr3**
Create a GatewayProcessor and modify the gateway application to use the GatewayProcessor.

Testing
====

1. Building all the OSGi docker images
2. Deploying the prerequisite and Corda helm charts.
3. Checking the Gateway has started in the Gateway container logs:
```
10:02:30.557 [lifecycle-coordinator-0] INFO  net.corda.processors.p2p.gateway.internal.GatewayProcessorImpl - Gateway processor is UP
```

I also tested the P2P deployment. Sending 100 messages between two counterparties. All 100 messages were successfully delivered.

### CODE REVIEW

File: processors/gateway-processor/src/main/kotlin/net/corda/processors/p2p/gateway/GatewayProcessor.kt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package net.corda.processors.p2p.gateway
+
+import net.corda.libs.configuration.SmartConfig
+
+/** The processor for a `FlowWorker`. */
```
Comment (dimosr) [inline processors/gateway-processor/src/main/kotlin/net/corda/processors/p2p/gateway/GatewayProcessor.kt]: something to remove?

File: processors/gateway-processor/src/main/kotlin/net/corda/processors/p2p/gateway/internal/GatewayProcessorImpl.kt [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        lifecycleCoordinator.postEvent(BootConfigEvent(bootConfig, useStubComponents))
+    }
+
+    override fun stop() {
+        logger.info("Gateway processor stopping.")
+        lifecycleCoordinator.stop()
+    }
+
+    private fun eventHandler(event: LifecycleEvent, coordinator: LifecycleCoordinator) {
+        logger.debug { "Link manager received event $event." }
```
Comment (dimosr) [inline processors/gateway-processor/src/main/kotlin/net/corda/processors/p2p/gateway/internal/GatewayProcessorImpl.kt]: to be removed?

**Review by dimosr — COMMENTED** [review]
LGTM - just two things that are probably left over accidentally.

File: processors/gateway-processor/src/main/kotlin/net/corda/processors/p2p/gateway/GatewayProcessor.kt (hunk 1 above)
Comment (williamvigorr3) [reply processors/gateway-processor/src/main/kotlin/net/corda/processors/p2p/gateway/GatewayProcessor.kt]: I updated this.

**Review by dimosr — APPROVED** [review]

**Pull request merged**